### PR TITLE
InputNumber ignores native attribute maxlength #5058

### PR DESCRIPTION
**Pull request by 40oleg**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behavior?

native maxlength just ignored

Closes # [](https://github.com/taiga-family/taiga-ui/issues/5058)

## What is the new behavior?

native maxlength works

### CODE REVIEW

File: projects/kit/components/input-number/input-number.component.ts [hunk 1]
```diff
@@ -163,6 +163,12 @@ export class TuiInputNumberComponent
     }
 
     get calculatedMaxLength(): number {
+        const nativeMaxLength = this.textfield?.nativeFocusableElement?.maxLength;
+
+        if (nativeMaxLength !== -1 && nativeMaxLength !== undefined) {
+            return nativeMaxLength;
+        }
```
Comment (splincode) [inline projects/kit/components/input-number/input-number.component.ts:170]: could you add unit tests?

**Comment by 40oleg** [comment]
I need to clarify something. One of the test failed. It just checks default "maxlength" value of input. But for some reason, the default "maxlength" value when running tests is different from the default "maxlength" value when running the application in the browser.
```ts
const el = document.createElement(`input`);
const out = el.maxLength;
console.log(out)
```
Another words, if you run code above in console of chrome you will get -1, but if you run it in .spec.ts file you will get 524228 (this is kind of max hard limit)
Apparently, something is wrong with runtime that was used to run the tests.
What should I do with this test? I can just hardcode this number in condition so test completes successfully, but maybe there is a better solution.
What do you think? @splincode

(found only one link for the kind of similar issue but it was dated 2011 https://bugs.webkit.org/show_bug.cgi?id=44883)

**Comment by splincode** [comment]
@40oleg what if we mock input reference as `const input = jest.fn();`?

File: projects/kit/components/input-number/input-number.component.ts [hunk 2]
```diff
@@ -163,6 +163,12 @@ export class TuiInputNumberComponent
     }
 
     get calculatedMaxLength(): number {
+        const nativeMaxLength = this.textfield?.nativeFocusableElement?.maxLength;
+
+        if (nativeMaxLength !== -1 && nativeMaxLength !== undefined) {
```
Comment (nsbarsukov) [inline projects/kit/components/input-number/input-number.component.ts:168]: [Unit tests / kit](https://github.com/taiga-family/taiga-ui/actions/runs/5871927108/job/15922430479?pr=5119) job has failed with the following error:

```console
 FAIL  projects/kit/components/input-number/test/input-number.component.spec.ts
  ● InputNumber › maxlength is set to 23 by default (18 digits + 5 default separators)

    expect(received).toBe(expected) // Object.is equality

    Expected: "23"
    Received: "524288"

      335 |         const nativeInput = getNativeInput()!.nativeElement;
      336 |
    > 337 |         expect(nativeInput.getAttribute(`maxlength`)).toBe(`23`);
          |                                                       ^
      338 |     });
      339 |
      340 |     describe(`When decimal === always`, () => {

      at projects/kit/components/input-number/test/input-number.component.spec.ts:337:55
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:409:30)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:3830:43)
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:408:56)
      at Zone.Object.<anonymous>.Zone.run (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:169:47)
      at Object.wrappedFunc (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:4330:34)
```

**Comment by nsbarsukov** [comment]
The same issue will be solved in this PR:

* https://github.com/taiga-family/taiga-ui/pull/5307

**Pull request closed without merging**